packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -417,6 +458,7 @@ Hardhat Network's forking functionality only works with blocks from at least spu
     }
 
     await this._resetNextBlockTimestamp();
+    this._resetUserProvidedNextBlockBaseFeePerGas();
```

Comment:
Unrelated to this PR, but why is `_resetNextBlockTimestamp` async? Worth fixing it here (assuming it's a mistake) just because the inconsistency is more obvious after this change.

Reply:
Done